examples/text/text_backspace_input.c

```diff
@@ -0,0 +1,129 @@
+// Program main entry point
+//------------------------------------------------------------------------------------
+int main(void)
+{
+    // Initialization
+    //--------------------------------------------------------------------------------------
+    const int screenWidth = 800;
+    const int screenHeight = 450;
+
+    InitWindow(screenWidth, screenHeight, "raylib [text] example - backspace input");
+
+    char textField[MAX_INPUT_CHARS + 1] = "\0";
+    int usedSize = 0;
+
+    SetTargetFPS(1);
+    //--------------------------------------------------------------------------------------
+
+    // Main game loop
+    while (!WindowShouldClose())    // Detect window close button or ESC key
+    {
+        // Update
+        //----------------------------------------------------------------------------------
+
+        int letter = GetCharPressed();
+        int key = GetKeyPressed();
+
+        while (key > 0 || letter > 0)
+        {
+
+            // ingest (key) backspace
+            // if posable delete one character from the string
+            if (key == KEY_BACKSPACE && usedSize > 0)
+            {
+                textField[usedSize - 1] = 0;
+                usedSize--;
+                key = GetKeyPressed();
+            }
+
+            // ingest (key) non character
+            else if (!isKeyChar(key))
+            {
+                key = GetKeyPressed();
+            }
+
+            // ingest (letter) and (key)
+            // if possible wright ascii character to the string
+            else if ((letter > 0) && (letter >= 32) && (letter <= 125) && (usedSize < MAX_INPUT_CHARS))
+            {
+                textField[usedSize] = (char) letter;
+                textField[usedSize + 1] = 0;
+                usedSize++;
+                letter = GetCharPressed();
+                key = GetKeyPressed();
+            }
+
+            // ingest (letter) and (key)
+            else
+            {
+                letter = GetCharPressed();
+                key = GetKeyPressed();
+            }
+        }
+
+
+        // Draw
+        //----------------------------------------------------------------------------------
+        BeginDrawing();
+
+            ClearBackground(RAYWHITE);
+
+            // TODO: Draw everything that requires to be drawn at this point:
+
+            DrawText(textField, 40, 200, 20, BLACK);  // Example
+
+        EndDrawing();
+        //----------------------------------------------------------------------------------
+    }
+
+    // De-Initialization
+    //--------------------------------------------------------------------------------------
+
+    CloseWindow();        // Close window and OpenGL context
+    //--------------------------------------------------------------------------------------
+
+    return 0;
+}
+
+bool isKeyChar(int key)
+{
+    switch (key)
+    {
+    case KEY_SPACE:
+    case KEY_APOSTROPHE:
+    case KEY_COMMA...KEY_NINE:
```

Comment:
This ... feature seems non-standard, it may not work on all compilers (likely visual studio)
I don't have it available to try but it's probaby worth removing either way
```
bruh.c:119:5: warning: range expressions in switch statements are non-standard [-Wpedantic]
  119 |     case KEY_COMMA...KEY_NINE:
      |     ^~~~
```

Reply:
@Peter0x44 @benev0 I don't understand how `IsKeyChar` is an improvement on using `ctypes.h` and `isprint(c)` or some other suitable combination of standard character classification functions.

Reply:
@orcmid Your advice is terrible. Never give it again, you are inviting bugs.
ctypes.h itself is a horrific footgun and any code you see including it is probably defective.
I ban it outright.

1) it's affected by locale (hopefully self-evident why this is bad)
2) it's designed for use with fgetc, with a domain of unsigned char + EOF. any negative inputs aside from EOF are UB,
So:
```c
char* x = ...;
if (isprint(x[0])) ...
```
is straight up incorrect.

I could rant even longer about it but I hope you get the message. Ban any use of ctype.h from your code immediately.

Reply:
I have updated the function to use bitwise operations.

@orcmid 
As for `isprint(int)` I believe the functionality is different. `isprint(int)` takes in a character code while `isKeyChar(int)` takes in a key code, so although keycodes mostly directly correspond, `isPrint(int)` does not appear to take into account numpad input. Do note that I have recently added numpad keys to fix a crash, so I believe you where correct that these where interchangeable functions.